storage/innobase/handler/ha_innodb.cc

```diff
@@ -7847,7 +7847,7 @@ ha_innobase::write_row(
 		m_prebuilt->autoinc_error = DB_SUCCESS;
 
 #ifdef WITH_WSREP
-		wsrep_auto_inc_inserted = trx->is_wsrep()
+		wsrep_auto_inc_inserted = wsrep_on(m_user_thd)
 			&& wsrep_drupal_282555_workaround
 			&& table->next_number_field->val_int() == 0;
```

Comment:
This seems to introduce a function call every time when there is an `INSERT` of a row into a table that contains an `AUTO_INCREMENT` column.

Can we at least change the order of the arguments, so that we will avoid paying the extra penalty?
```c++
#ifdef WITH_WSREP
		wsrep_auto_inc_inserted = trx->is_wsrep()
		wsrep_auto_inc_inserted = UNIV_UNLIKELY(wsrep_drupal_282555_workaround)
			&& wsrep_on(m_user_thd)
			&& table->next_number_field->val_int() == 0;
#endif
```
The default value of that global parameter is `false`.

Reply:
I agree because default value for wsrep_drupal_282555_workaround is false (and naturally I hope it is not used if wsrep_on != ON.